File: src/main/java/com/restaurant/entity/Commande.java
package com.restaurant.entity;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Commande extends BaseEntity {
    private Client client;
    private List<CommandeItem> items;
    private EtatCommande etat;
    private String typeLivraison; // "SUR_PLACE", "A_EMPORTER", "LIVRAISON"
    private LocalDateTime dateCommande;
    private LocalDateTime dateLivraison;
    private Zone zone;
    private Livreur livreur;
    private String adresseLivraison;
    private double total;
    private boolean payee;
    
    public Commande() {
        this.items = new ArrayList<>();
        this.etat = EtatCommande.EN_ATTENTE;
        this.dateCommande = LocalDateTime.now();
        this.payee = false;
    }
    
    public Commande(Client client, String typeLivraison) {
        this();
        this.client = client;
        this.typeLivraison = typeLivraison;
    }
    
    public Client getClient() {
        return client;
    }
    
    public void setClient(Client client) {
        this.client = client;
    }
    
    public List<CommandeItem> getItems() {
        return items;
    }
    
    public void setItems(List<CommandeItem> items) {
        this.items = items;
        calculerTotal();
    }
    
    public void addItem(CommandeItem item) {
        this.items.add(item);
        calculerTotal();
    }
    
    public EtatCommande getEtat() {
        return etat;
    }
    
    public void setEtat(EtatCommande etat) {
        this.etat = etat;
    }
    
    public String getTypeLivraison() {
        return typeLivraison;
    }
    
    public void setTypeLivraison(String typeLivraison) {
        this.typeLivraison = typeLivraison;
    }
    
    public LocalDateTime getDateCommande() {
        return dateCommande;
    }
    
    public void setDateCommande(LocalDateTime dateCommande) {
        this.dateCommande = dateCommande;
    }
    
    public LocalDateTime getDateLivraison() {
        return dateLivraison;
    }
    
    public void setDateLivraison(LocalDateTime dateLivraison) {
        this.dateLivraison = dateLivraison;
    }
    
    public Zone getZone() {
        return zone;
    }
    
    public void setZone(Zone zone) {
        this.zone = zone;
    }
    
    public Livreur getLivreur() {
        return livreur;
    }
    
    public void setLivreur(Livreur livreur) {
        this.livreur = livreur;
    }
    
    public String getAdresseLivraison() {
        return adresseLivraison;
    }
    
    public void setAdresseLivraison(String adresseLivraison) {
        this.adresseLivraison = adresseLivraison;
    }
    
    public double getTotal() {
        return total;
    }
    
    public void setTotal(double total) {
        this.total = total;
    }
    
    public boolean isPayee() {
        return payee;
    }
    
    public void setPayee(boolean payee) {
        this.payee = payee;
    }
    
    private void calculerTotal() {
        total = items.stream()
                     .mapToDouble(CommandeItem::getSousTotal)
                     .sum();
        
        if (typeLivraison.equals("LIVRAISON") && zone != null) {
            total += zone.getPrixLivraison();
        }
    }
    
    public String getResume() {
        return "Commande #" + getId() + 
               " - Client: " + client.getNomComplet() + 
               " - Total: " + total + " FCFA" +
               " - État: " + etat;
    }
    
    @Override
    public String toString() {
        return "Commande{" +
               "id=" + getId() +
               ", client=" + client.getNomComplet() +
               ", nbItems=" + items.size() +
               ", etat=" + etat +
               ", typeLivraison='" + typeLivraison + '\'' +
               ", total=" + total +
               ", payee=" + payee +
               ", archived=" + isArchived() +
               '}';
    }
}

File: src/Entity/Commande.php
<?php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CommandeRepository::class)]
#[ORM\Table(name: 'commandes')]
class Commande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Client::class, inversedBy: 'commandes')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Client $client = null;

    #[ORM\Column(length: 20)]
    private string $etat = EtatCommande::EN_ATTENTE;

    #[ORM\Column(length: 20)]
    private ?string $typeLivraison = null;

    #[ORM\Column(type: 'datetime_immutable')]
    private \DateTimeImmutable $dateCommande;

    #[ORM\Column(type: 'datetime_immutable', nullable: true)]
    private ?\DateTimeImmutable $dateLivraison = null;

    // #[ORM\ManyToOne(targetEntity: Zone::class)]
    #[ORM\ManyToOne(targetEntity: Zone::class, inversedBy: 'commandes')]
    private ?Zone $zone = null;

    // #[ORM\ManyToOne(targetEntity: Livreur::class)]
    #[ORM\ManyToOne(targetEntity: Livreur::class, inversedBy: 'commandes')]
    private ?Livreur $livreur = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $adresseLivraison = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $total = null;

    #[ORM\Column(type: 'boolean')]
    private bool $payee = false;

    #[ORM\Column(type: 'boolean')]
    private bool $archived = false;

    #[ORM\Column(type: 'datetime_immutable')]
    private \DateTimeImmutable $createdAt;

    #[ORM\Column(type: 'datetime_immutable')]
    private \DateTimeImmutable $updatedAt;

    #[ORM\OneToMany(mappedBy: 'commande', targetEntity: CommandeItem::class, cascade: ['persist', 'remove'])]
    private Collection $commandeItems;

    #[ORM\OneToOne(mappedBy: 'commande', targetEntity: Paiement::class, cascade: ['persist', 'remove'])]
    private ?Paiement $paiement = null;

    public function __construct()
    {
        $this->dateCommande = new \DateTimeImmutable();
        $this->createdAt = new \DateTimeImmutable();
        $this->updatedAt = new \DateTimeImmutable();
        $this->commandeItems = new ArrayCollection();
    }

    // Getters and setters
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getClient(): ?Client
    {
        return $this->client;
    }

    public function setClient(?Client $client): self
    {
        $this->client = $client;
        return $this;
    }

    public function getEtat(): string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        if (!in_array($etat, EtatCommande::getValues())) {
            throw new \InvalidArgumentException("État invalide: $etat");
        }
        $this->etat = $etat;
        return $this;
    }

    public function getEtatLabel(): string
    {
        return EtatCommande::getLabel($this->etat);
    }

    public function getTypeLivraison(): ?string
    {
        return $this->typeLivraison;
    }

    public function setTypeLivraison(string $typeLivraison): self
    {
        $this->typeLivraison = $typeLivraison;
        return $this;
    }

    public function getDateCommande(): \DateTimeImmutable
    {
        return $this->dateCommande;
    }

    public function setDateCommande(\DateTimeImmutable $dateCommande): self
    {
        $this->dateCommande = $dateCommande;
        return $this;
    }

    public function getDateLivraison(): ?\DateTimeImmutable
    {
        return $this->dateLivraison;
    }

    public function setDateLivraison(?\DateTimeImmutable $dateLivraison): self
    {
        $this->dateLivraison = $dateLivraison;
        return $this;
    }

    public function getZone(): ?Zone
    {
        return $this->zone;
    }

    public function setZone(?Zone $zone): self
    {
        $this->zone = $zone;
        return $this;
    }

    public function getLivreur(): ?Livreur
    {
        return $this->livreur;
    }

    public function setLivreur(?Livreur $livreur): self
    {
        $this->livreur = $livreur;
        return $this;
    }

    public function getAdresseLivraison(): ?string
    {
        return $this->adresseLivraison;
    }

    public function setAdresseLivraison(?string $adresseLivraison): self
    {
        $this->adresseLivraison = $adresseLivraison;
        return $this;
    }

    public function getTotal(): ?string
    {
        return $this->total;
    }

    public function setTotal(string $total): self
    {
        $this->total = $total;
        return $this;
    }

    public function isPayee(): bool
    {
        return $this->payee;
    }

    public function setPayee(bool $payee): self
    {
        $this->payee = $payee;
        return $this;
    }

    public function isArchived(): bool
    {
        return $this->archived;
    }

    public function setArchived(bool $archived): self
    {
        $this->archived = $archived;
        return $this;
    }

    public function getCreatedAt(): \DateTimeImmutable
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeImmutable $createdAt): self
    {
        $this->createdAt = $createdAt;
        return $this;
    }

    public function getUpdatedAt(): \DateTimeImmutable
    {
        return $this->updatedAt;
    }

    public function setUpdatedAt(\DateTimeImmutable $updatedAt): self
    {
        $this->updatedAt = $updatedAt;
        return $this;
    }

    /**
     * @return Collection<int, CommandeItem>
     */
    public function getCommandeItems(): Collection
    {
        return $this->commandeItems;
    }

    public function addCommandeItem(CommandeItem $commandeItem): self
    {
        if (!$this->commandeItems->contains($commandeItem)) {
            $this->commandeItems->add($commandeItem);
            $commandeItem->setCommande($this);
        }

        return $this;
    }

    public function removeCommandeItem(CommandeItem $commandeItem): self
    {
        if ($this->commandeItems->removeElement($commandeItem)) {
            // set the owning side to null (unless already changed)
            if ($commandeItem->getCommande() === $this) {
                $commandeItem->setCommande(null);
            }
        }

        return $this;
    }

    public function getPaiement(): ?Paiement
    {
        return $this->paiement;
    }

    public function setPaiement(?Paiement $paiement): self
    {
        // set the owning side of the relation if necessary
        if ($paiement !== null && $paiement->getCommande() !== $this) {
            $paiement->setCommande($this);
        }

        $this->paiement = $paiement;
        return $this;
    }

    // Méthodes utilitaires
    public function getTotalItems(): int
    {
        $total = 0;
        foreach ($this->commandeItems as $item) {
            $total += $item->getQuantite();
        }
        return $total;
    }

    public function __toString(): string
    {
        return sprintf('Commande #%d - %s', $this->id, $this->client?->getNom() ?? '');
    }
}